cmd/agent/app/snmp.go

```diff
@@ -293,7 +293,13 @@ func printValue(pdu gosnmp.SnmpPDU) error {
 	switch pdu.Type {
 	case gosnmp.OctetString:
 		b := pdu.Value.([]byte)
-		fmt.Printf("STRING: %s\n", string(b))
+		if !IsStringPrintable(b) {
+			var strValue string
+			strValue = fmt.Sprintf("%X", b)
+			fmt.Printf("Hex-STRING: %s %s %s %s %s %s\n", strValue[0:2], strValue[2:4], strValue[4:6], strValue[6:8], strValue[8:10], strValue[10:12])
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** I think we need to loop through the bytes since the number of bytes since we might have more than 6 elements.